analysis.md

```diff
@@ -0,0 +1,77 @@
+ofile = open(sys.argv[2], "w")
+
+encoded = ""
+
+for line in ifile:
+	encoded += line
+
+ofile.write(base64.b64decode(encoded))
+```
+After decoding, you'll notice that the resulting contents are still unreadable because they are gzipped. Decompress the contents using `gunzip` and you'll see an obfuscated powershell script with strange variable names meant to confuse readers. After cleaning up this mess through refactoring the code, there are many interesting things to note about malware persistence.
+
+The script uses various methods to keep the encoded third layer payload running on the victim machine regularly.
+
+## Layer 2
+
+### Storing the Payload
+If the victim has Powershell 3.0 or later, the script stores the encoded layer 3 payload inside `kernel32.dll`. Otherwise, the script adds a new property called `Path` to a key in the registry to store the payload. If the user has Administrator permissions, the payload will be stored in `HKLM:Software\Microsoft\Windows\CurrentVersion`. Otherwise, it will be stored in `HKCU:Software\Microsoft\Windows`.
+
+### Non-Admins: Registry Keys
+The script creates a file called `kernel32.vbs` and writes to it instructions to retrieve and execute the payload.
+
+The attacker modifies Run registry keys to cause `kernel.vbs` to run after waiting 30 minutes each time a user logs on. If the user has administrator permissions, the targeted key is `HKLM:Software\Microsoft\Windows\CurrentVersion\Run\`. Otherwise, the targeted key is `HKCU:Software\Microsoft\Windows\CurrentVersion\Run\`.
+
+### Admins: Creation of a New WMI Object
+If the user has administrator permissions, the script creates a permanent WMI event subscription that watches for a user logon, waits 30 minutes, then executes the layer 3 payload. At this point, the program has exhibited enough strange behavior for the SentinelAgent to identify it as malware.
+
+### Scheduled Task
+The script also creates a scheduled task to run the malware 30 minutes after a user logs on. The method used to start the malware differs only slightly depending on if the user has administrator permissions. If the user is a non-admin, `wscript` is used to execute the contents of `kernel32.vbs`. Otherwise, a Invoke-Expression (IEX) cmdlet is used.
+
+## Layer 3
+Now that we've discussed the various ways the attacker achieves persistence of the malware, lets find out what the layer 3 payload is responsible for. You'll notice yet another Base64 blob inside layer 2.
+
+You'll notice an attempt to thwart static analysis at the creation of a mutex inside the function `logic`, named `"SourceFireSux"`, a jab aimed at the security company. It may seem that the function exits prompty, but the boolean variable is set to `True` after the [construction of the mutex](https://msdn.microsoft.com/en-us/library/bwe34f1k(v=vs.110).aspx).
+
+```
+[bool]$var = $false;
+$mutex = New-Object System.Threading.Mutex($true, “SourceFireSux”, [ref] $var);        
+if (!$var)
```

Comment:
You can clean up some of the syntactic sugar for brevity, like taking out some newlines so it's just `if (!$var) { exit; }`